error: implicit declaration of function 'strdup' is invalid in C99

Hi, I'm the maintainer of xdotool in MacPorts. xdotool fails to build on macOS when using Xcode 12 or later. The error is:

```
xdo.c:1365:26: error: implicit declaration of function 'strdup' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  keyseq_copy = strptr = strdup(keyseq);
                         ^
```

This was reported to MacPorts here: https://trac.macports.org/ticket/61990

In Xcode 12, Apple changed implicit declaration of functions from a warning to an error, because Apple introduced Macs using ARM processors, and ARM processors use different calling conventions for variadic and non-variadic functions, so the compiler must know before you call a function (via an explicit declaration) what kind of function it is.

`strdup` is defined in string.h. xdo.c already includes string.h but it also defines `_XOPEN_SOURCE` to 500 if it is not already defined. If you don't define `_XOPEN_SOURCE`, then `strdup` is available, but if you do define `_XOPEN_SOURCE`, what you define it to determines what functions are available. My understanding is that on Linux, defining `_XOPEN_SOURCE` to at least 500 is enough to make `strdup` available, however on macOS if you define it at all then you must define it to at least 600 to make `strdup` available.

I am working around the problem for now by adding `-D_XOPEN_SOURCE=600` to CFLAGS when I compile but you may wish to fix it in your sources so that it can be compiled without such intervention. I don't know how you would like to fix it: you could either change all of your `#define _XOPEN_SOURCE 500` to `#define _XOPEN_SOURCE 600`—I don't know if that would have any adverse effects for any other operating systems—or you could define it conditionally, setting it to 600 on macOS (and other operating systems, if any, that require this) and 500 on others.

--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,5 @@
 *.o
+*.dylib
 *.so*
 *.1
 libxdo.pc

--- a/xdo.c
+++ b/xdo.c
@@ -7,7 +7,7 @@
  */
 
 #ifndef _XOPEN_SOURCE
-#define _XOPEN_SOURCE 500
+#define _XOPEN_SOURCE 600
 #endif /* _XOPEN_SOURCE */
 
 #include <sys/select.h>

